pkg/storage/minio/checker.go

```diff
@@ -8,12 +8,16 @@ import (
 	opentracing "github.com/opentracing/opentracing-go"
 )
 
-func (v *storageImpl) Exists(ctx context.Context, module, version string) bool {
+func (v *storageImpl) Exists(ctx context.Context, module, version string) (bool, error) {
 	sp, ctx := opentracing.StartSpanFromContext(ctx, "storage.minio.Exists")
 	defer sp.Finish()
 	versionedPath := v.versionLocation(module, version)
 	modPath := fmt.Sprintf("%s/go.mod", versionedPath)
 	_, err := v.minioClient.StatObject(v.bucketName, modPath, minio.StatObjectOptions{})
 
-	return err == nil
+	if minio.ToErrorResponse(err).Code == "NoSuchKey" {
+		return false, nil
+	}
+
+	return err == nil, err
```

Comment:
let's wrap this err with `errors.E` if it's not nil. I keep thinking if we should do it like `github.com/pkg/errors` where we can do something like `returns err == nil, errors.E(op, err)` and that if the err arg is nil, then we return nil. This makes writing the code easier and nicer, but it's very implicit. The current way is to just check `if err != nil` which is more writing, but clearer.